File: Sum_of_Segments.py
# This code is a translation from C++ to Python and uses standard input/output.
class FastIO:
    def __init__(self):
        import sys
        input = sys.stdin.read
        sys.stdin = open(0)  # Disable C I/O synchronization
        sys.stdout = open(1, 'w')  # Untie cin and cout for faster input/output

fast_io_setup = FastIO()  # This will be executed before main()

def main():
    n = int(input().strip())
    num = list(map(int, input().strip().split()))

    pre = [0] * (n + 1)
    for i in range(n):
        pre[i + 1] = pre[i] + num[i]

    sz = n * (n + 1) // 2
    sums = [0] * (sz + 1)
    idx = 1
    for l in range(n):
        for r in range(l, n):
            sums[idx] = sums[idx - 1] + (pre[r + 1] - pre[l])
            idx += 1

    q = int(input().strip())
    for _ in range(q):
        l, r = map(int, input().strip().split())
        print(sums[r] - sums[l - 1])

if __name__ == "__main__":
    main()

File: Sum_of_Segments.cpp
#pragma GCC optimize("O2")
#include <bits/stdc++.h>
using namespace std;
typedef long long ll;

// Global setup for fast I/O
struct FastIO {
    FastIO() {
        ios_base::sync_with_stdio(false); // Disable C I/O synchronization
        cin.tie(nullptr); // Untie cin and cout for faster input/output
    }
} fast_io_setup; // This will be executed before main()

int main() {
    ll n;
    cin >> n;
    vector<ll> num(n);
    for (ll i = 0; i < n; i++) {
        cin >> num[i];
    }

    vector<ll> pre(n + 1, 0);
    for (ll i = 0; i < n; i++) {
        pre[i + 1] = pre[i] + num[i];
    }

    ll sz = n * (n + 1) / 2;
    vector<ll> sums(sz + 1, 0);
    ll idx = 1;
    for (ll l = 0; l < n; l++) {
        for (ll r = l; r < n; r++) {
            sums[idx] = sums[idx - 1] + (pre[r + 1] - pre[l]);
            idx++;
        }
    }

    ll q;
    cin >> q;
    while (q--) {
        ll l, r;
        cin >> l >> r;
        cout << sums[r] - sums[l - 1] << "\n";
    }
}
